### PR TITLE
convert WITH clause to xgboost params

### PR DESCRIPTION
**Pull request by Yancey0623**
a part work of #749

### CODE REVIEW

File: sql/codegen_xgboost.go [hunk 1]
```diff
@@ -14,41 +14,109 @@
...
+		xgbTrainConfig: *resolveTrainCfg(attrs),
+		Save:           pr.save,
+	}
+
+	params, err := resolveParamsCfg(attrs)
+	if err != nil {
+		return nil, err
+	}
+
+	// fill learning targe
```
Comment (typhoonzero) [inline sql/codegen_xgboost.go]: typo: targe

File: sql/codegen_xgboost.go [hunk 2]
```diff
@@ -14,41 +14,109 @@
...
+		Estimator: Estimator{
+			IsTrain:              pr.train,
+			TrainingDatasetSQL:   training,
+			ValidationDatasetSQL: validation,
+		},
+		xgbTrainConfig: *resolveTrainCfg(attrs),
+		Save:           pr.save,
+	}
+
+	params, err := resolveParamsCfg(attrs)
```
Comment (typhoonzero) [inline sql/codegen_xgboost.go:104]: `resolveParamsCfg` must run after `resolveTrainCfg` to resolve all other attributes, maybe add some comment here as hints.

**Review by typhoonzero — COMMENTED** [review]
LGTM with minor comments.

File: sql/codegen_xgboost.go (hunk 2 above)
Comment (Yancey0623) [reply sql/codegen_xgboost.go:104]: `resolveParamsCfg ` only resolve the keys without prefix, maybe the order is fine?

File: sql/codegen_xgboost.go (hunk 1 above)
Comment (Yancey0623) [reply sql/codegen_xgboost.go]: thx, done.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**